src/main/java/com/github/manolo8/darkbot/gui/players/PlayerEditor.java

```diff
@@ -4,29 +4,71 @@
 import com.github.manolo8.darkbot.gui.utils.Popups;
 import com.github.manolo8.darkbot.gui.utils.SearchField;
-import com.github.manolo8.darkbot.gui.utils.UIUtils;
 import com.github.manolo8.darkbot.utils.I18n;
+import eu.darkbot.api.events.EventHandler;
+import eu.darkbot.api.events.Listener;
+import eu.darkbot.api.game.entities.Player;
+import eu.darkbot.api.managers.EntitiesAPI;
+import eu.darkbot.api.managers.EventBrokerAPI;
 import net.miginfocom.swing.MigLayout;
 
 import javax.swing.*;
-import java.awt.event.ActionEvent;
-import java.util.Comparator;
-import java.util.List;
-import java.util.Locale;
+import java.util.*;
 
-public class PlayerEditor extends JPanel {
+import static java.util.stream.Collectors.toList;
+
+public class PlayerEditor extends JPanel implements Listener {
     private final JList<PlayerInfo> playerInfoList;
     private final DefaultListModel<PlayerInfo> playersModel;
+    private final List<PlayerInfo> nearbyPlayerList = new ArrayList<>();
 
     private final PlayerManager playerManager;
     private final SearchField sf;
     private final PlayerTagManager tagEditor;
     protected Main main;
 
+    @EventHandler
+    public void onEntityPlayerAdd(EntitiesAPI.EntityCreateEvent event) {
+        if (event.getEntity() instanceof Player) {
+            if (playersModel != null && main.config.PLAYER_INFOS != null && main.config.PLAYER_INFOS.values()
+                    .stream().noneMatch(a -> a.toString()
+                            .contains(((Player) event.getEntity()).getEntityInfo().getUsername()))) {
+                nearbyPlayerList.add(new PlayerInfo(((Player) event.getEntity())
+                        .getEntityInfo().getUsername(), event.getEntity().getId()));
+
+                SwingUtilities.invokeLater(() -> playersModel.
+                        addElement(new PlayerInfo(((Player) event.getEntity())
+                                .getEntityInfo().getUsername(), event.getEntity().getId())));
+                System.out.println("added " + ((Player) event.getEntity()).getEntityInfo().getUsername());
+            } else {
+                System.out.println("in list yet " + ((Player) event.getEntity()).getEntityInfo().getUsername());
+            }
+        }
+    }
+
+    @EventHandler
+    public void onEntityPlayerRemove(EntitiesAPI.EntityRemoveEvent event) {
+        if (event.getEntity() instanceof Player) {
+            if (playersModel != null && main.config.PLAYER_INFOS != null && main.config.PLAYER_INFOS.values()
+                    .stream()
+                    .noneMatch(a -> a.toString()
+                            .contains(((Player) event.getEntity()).getEntityInfo().getUsername()))) {
+                nearbyPlayerList.stream()
+                        .filter(a -> a.toString().contains(((Player) event.getEntity()).getEntityInfo().getUsername()))
+                        .collect(toList()).forEach(nearbyPlayerList::remove);
```

Comment:
You're iterating the player list, converting to string, checking for contains which could return multiple results, and removing all of them. The code is wrong in several ways:

 - If there's a player named `PlayerABCDE` and another player named `Player`, when `Player` gets out of sight, both are removed.
 - You're converting a PlayerInfo to string to check for the username, that to string contains even the tag because it's used for filtering, and regardless, you should never rely on toString of an object for your logic, as that is prone to changing.
 - You're collecting all results to a list, causing another list allocation
 
 You can just use removeIf:
 ```suggestion
                nearbyPlayerList.removeIf(pl -> pl.getId() == event.getEntity().getId());
```

Keep in mind however, none of this is relevant because you should be doing this in the swing event thread to avoid race conditions on nearbyPlayerList, as i stated in player add. See comment below for how to easily do it in the swing event thread.